client/reader/controller.js

```diff
@@ -63,10 +62,9 @@ pageNotifier( function removeFullPostOnLeave( newContext, oldContext ) {
 		return;
 	}
 
-	const fullPostViewPrefix = '/read/post/';
+	const fullPostViewRegex = /\/read\/(blogs|feeds)\/([0-9]+)\/posts\/([0-9]+)/i;
 
-	if ( startsWith( oldContext.path, fullPostViewPrefix ) &&
-		! startsWith( newContext.path, fullPostViewPrefix ) ) {
+	if ( oldContext.path.match( fullPostViewRegex ) && ! newContext.path.match( fullPostViewRegex ) ) {
```

Comment:
as long as you're in here... 

could you work in https://github.com/Automattic/wp-calypso/commit/a32b376d8d55caaf233c6678e960314d3a45c5f0#diff-b83a4548b9f95eff8c33bf12ec2bcd30 ? It turns out this method is busted if you start on full post and then go directly to the editor using a normal link.